Marlin/Configuration.h

```diff
@@ -808,10 +808,21 @@
   #define ABL_GRID_POINTS_Y ABL_GRID_POINTS_X
 
   // Set the boundaries for probing (where the probe can reach).
-  #define LEFT_PROBE_BED_POSITION 15
-  #define RIGHT_PROBE_BED_POSITION 170
-  #define FRONT_PROBE_BED_POSITION 20
-  #define BACK_PROBE_BED_POSITION 170
+  #define PROBE_BED_MARGIN 5
+  #if X_PROBE_OFFSET_FROM_EXTRUDER < 0
+    #define LEFT_PROBE_BED_POSITION PROBE_BED_MARGIN
+    #define RIGHT_PROBE_BED_POSITION (X_MAX_POS - PROBE_BED_MARGIN + X_PROBE_OFFSET_FROM_EXTRUDER)
+  #else
+    #define LEFT_PROBE_BED_POSITION (X_PROBE_OFFSET_FROM_EXTRUDER + PROBE_BED_MARGIN)
+    #define RIGHT_PROBE_BED_POSITION (X_MAX_POS - PROBE_BED_MARGIN)
+  #endif
+  #if X_PROBE_OFFSET_FROM_EXTRUDER < 0
+    #define FRONT_PROBE_BED_POSITION PROBE_BED_MARGIN
+    #define BACK_PROBE_BED_POSITION (Y_MAX_POS - PROBE_BED_MARGIN + Y_PROBE_OFFSET_FROM_EXTRUDER)
+  #else
+    #define FRONT_PROBE_BED_POSITION (Y_PROBE_OFFSET_FROM_EXTRUDER + PROBE_BED_MARGIN)
+    #define BACK_PROBE_BED_POSITION (Y_MAX_POS - PROBE_BED_MARGIN)
+  #endif
```

Comment:
Generally all good suggestions. I think the auto-assigning of these values should be moved into `Conditionals_POST.h`, in this manner:
```cpp
#ifndef LEFT_PROBE_BED_POSITION
  #define LEFT_PROBE_BED_POSITION  (X_PROBE_OFFSET_FROM_EXTRUDER <= MIN_PROBE_EDGE ? X_MIN_POS + MIN_PROBE_EDGE : X_MIN_POS + X_PROBE_OFFSET_FROM_EXTRUDER)
#endif
```
...and so on. Then, comment out the lines in `Configuration.h` and add documentation that these values are automatically set and only need to be changed under special conditions. Some of the example configurations might provide overrides. Most of them will probably be able to leave them undefined.